test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java

```diff
@@ -360,17 +362,21 @@ public void sourceTags() throws Exception {
         assertThat("incorrect object count", res2.getObjects().size(), is(1));
         TestCommon.compare(res2.getObjects().get(0), expected1);
     }
-    
+
     @Test
     public void narrativeDecoration() throws Exception {
```

Comment:
Ideally, these tests require clearing of cache in between tests, as the values cached in AccessGroupCache and NarrativeInfoCache in one test affects another. I have added some dummy workspaces in tests in order to fix that dependency.

Please let me know if they can be left as they are, or would you recommend clearing cache.

One way to make sure that the cache gets cleared is to give the minimum cachelife time, for testing. The cache values are initialized in KBaseSearchEngineServer(). They are now using constants, and can possibly made to get these values from config. That may not be a good option, since it requires different values for testing and deployment.

Reply:
I think we should retain the creation of separate workspaces for each test for maintaining separation. How about adding a @before method that creates a new instance of searchCli? Would this create a new set of cache instances? I don't think this is a time consuming step that will slow down the tests.

Reply:
The server needs to be restarted in order to clear the cache. Have modified so that the server gets restarted after every test, and have made the corresponding changes in the tests.